server/public/model/search_params.go

```diff
@@ -11,7 +11,7 @@ import (
 )
 
 var searchTermPuncStart = regexp.MustCompile(`^[^\pL\d\s#"]+`)
-var searchTermPuncEnd = regexp.MustCompile(`[^\pL\d\s*"]+$`)
+var searchTermPuncEnd = regexp.MustCompile(`[^\pL\p{Thai}\d\s*"]+$`)
```

Comment:
Now that I'm looking into this, it looks like this regex might remove things like accents in other languages as well. Could we maybe make this a bit broader to also support other languages too?

Looking at the list of Unicode categories on [this website](https://www.regular-expressions.info/unicode.html#prop), it looks like `\p{M}` might work for Thai and other languages as well